zebROS_ws/src/behaviors/src/2023_updown_elevater_server.cpp

```diff
@@ -0,0 +1,214 @@
+    _ddr.registerVariable<double>("VERTICAL_CONE_low_node", &_game_piece_lookup[elevater_ns::VERTICAL_CONE][elevater_ns::LOW_NODE], "", 0, 4);
+    _ddr.registerVariable<double>("VERTICAL_CONE_middle_node", &_game_piece_lookup[elevater_ns::VERTICAL_CONE][elevater_ns::MIDDLE_NODE], "", 0, 4);
+    _ddr.registerVariable<double>("VERTICAL_CONE_high_node", &_game_piece_lookup[elevater_ns::VERTICAL_CONE][elevater_ns::HIGH_NODE], "", 0, 4);
+
+    _ddr.registerVariable<double>("BASE_TOWARDS_US_CONE_intake", &_game_piece_lookup[elevater_ns::BASE_TOWARDS_US_CONE][elevater_ns::INTAKE], "", 0, 4);
+    _ddr.registerVariable<double>("BASE_TOWARDS_US_CONE_low_node", &_game_piece_lookup[elevater_ns::BASE_TOWARDS_US_CONE][elevater_ns::LOW_NODE], "", 0, 4);
+    _ddr.registerVariable<double>("BASE_TOWARDS_US_CONE_middle_node", &_game_piece_lookup[elevater_ns::BASE_TOWARDS_US_CONE][elevater_ns::MIDDLE_NODE], "", 0, 4);
+    _ddr.registerVariable<double>("BASE_TOWARDS_US_CONE_high_node", &_game_piece_lookup[elevater_ns::BASE_TOWARDS_US_CONE][elevater_ns::HIGH_NODE], "", 0, 4);
+
+    _ddr.registerVariable<double>("BASE_AWAY_US_CONE_intake", &_game_piece_lookup[elevater_ns::BASE_AWAY_US_CONE][elevater_ns::INTAKE], "", 0, 4);
+    _ddr.registerVariable<double>("BASE_AWAY_US_CONE_low_node", &_game_piece_lookup[elevater_ns::BASE_AWAY_US_CONE][elevater_ns::LOW_NODE], "", 0, 4);
+    _ddr.registerVariable<double>("BASE_AWAY_US_CONE_middle_node", &_game_piece_lookup[elevater_ns::BASE_AWAY_US_CONE][elevater_ns::MIDDLE_NODE], "", 0, 4);
+    _ddr.registerVariable<double>("BASE_AWAY_US_CONE_high_node", &_game_piece_lookup[elevater_ns::BASE_AWAY_US_CONE][elevater_ns::HIGH_NODE], "", 0, 4);
+
+    _ddr.publishServicesTopics();
+    _as.start();
+    ElevaterINFO("Started Elevater Action server");
+  }
+
+  ~ElevaterAction2023(void)
+  {
+  }
+
+  void print_map() {
+    for(const auto& elem : _game_piece_lookup)
+    {
+      std::cout << elem.first << "\n";
+      for (const auto& sub_elem : elem.second) {
+        std::cout << sub_elem.first << " " << sub_elem.second << "\n";
+      }
+      std::cout << "\n\n";
+    }
+  }
+
+  void executeCB(const behavior_actions::Elevater2023GoalConstPtr &goal)
+  {
+    ros::spinOnce();
+    // select piece, nice synatax makes loading params worth it
+    double position = _game_piece_lookup[goal->piece][goal->mode];
+
+    // apply offset
+    position += _position_offset;
+    ElevaterINFO("Moving a " << piece_to_string[goal->piece] << " to the position " << mode_to_string[goal->mode] << " and the ELEVATOR to the position=" << position << " meters");
+
+    assert(position >= 0); // probably done in elevator server also 
+    
+    controllers_2023_msgs::ElevatorSrv req;
+    req.request.position = position;
+
+    if (_elevator_srv.call(req)) {
+      ElevaterINFO("Succeeded moving elevator!");
+      _feedback.success = true;
```

Comment:
These can be local rather than member vars, since there's nothing requiring them to be persistent across calls to executeCB